tests/forms_tests/tests/test_formsets.py

```diff
@@ -1855,6 +1855,21 @@ class FileForm(Form):
 
         self.assertTrue(formset_factory(FileForm, extra=0)().is_multipart())
 
+    def test_empty_formset_with_empty_permitted(self):
+        """empty_form ignores empty_permitted argument."""
+        formset = ArticleFormSet(form_kwargs={"empty_permitted": True})
+
+        self.assertHTMLEqual(
+            str(formset.empty_form),
+            "<div>"
+            '<label for="id_form-__prefix__-title">Title:</label>'
+            '<input type="text" name="form-__prefix__-title" id="id_form-__prefix__-title">'
+            "</div>"
+            "<div>"
+            '<label for="id_form-__prefix__-pub_date">Pub date:</label>'
+            '<input type="text" name="form-__prefix__-pub_date" id="id_form-__prefix__-pub_date">'
+            "</div>",
+        )
```

Comment:
IMO checking how `empty_form` is rendered is not necessary. It's enough to check `empty_permitted`, e.g.
```suggestion
        self.assertIs(formset.empty_form.empty_permitted, True)

```

Reply:
ohh okay !